odoo/addons/base/models/res_lang.py

```diff
@@ -6,17 +6,45 @@
 import locale
 import logging
 import re
-from operator import itemgetter
+from typing import Any, Literal
 
 from odoo import api, fields, models, tools, _
 from odoo.exceptions import UserError, ValidationError
+from odoo.tools import OrderedSet, frozendict
 
 _logger = logging.getLogger(__name__)
 
 DEFAULT_DATE_FORMAT = '%m/%d/%Y'
 DEFAULT_TIME_FORMAT = '%H:%M:%S'
 
 
+class LangData(frozendict):
+    """ A dict-like class which can access field value like a res.lang record
+    Note: This data class cannot store data for fields with the same name as
+    `dict` methods e.g. dict.keys
+    """
+    __slots__ = ()
+
+    def __bool__(self) -> bool:
+        return bool(self.id)
+
+    def __getattr__(self, field: str) -> Any:
```

Comment:
Nitpicking: the official API uses `name`:
```suggestion
    def __getattr__(self, name: str) -> Any:
```